runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkPortableClientRunner.java

```diff
@@ -0,0 +1,250 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.flink;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.apache.beam.model.pipeline.v1.RunnerApi;
+import org.apache.beam.runners.fnexecution.environment.ProcessManager;
+import org.apache.beam.runners.fnexecution.jobsubmission.JobInvocation;
+import org.apache.beam.runners.fnexecution.jobsubmission.JobInvoker;
+import org.apache.beam.runners.fnexecution.jobsubmission.PortablePipelineResult;
+import org.apache.beam.runners.fnexecution.jobsubmission.PortablePipelineRunner;
+import org.apache.beam.runners.fnexecution.provisioning.JobInfo;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.ListeningExecutorService;
+import org.apache.flink.api.common.time.Deadline;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.client.program.OptimizerPlanEnvironment;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.kohsuke.args4j.Option;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Launch a Beam pipeline via external SDK driver program in the Flink {@link
+ * OptimizerPlanEnvironment}.
+ *
+ * <p>This entry point can be used to execute an external program to launch a Beam pipeline within
+ * the Flink {@link OptimizerPlanEnvironment}, which is present when running a job via the REST API.
+ *
+ * <p>Designed for non-interactive Flink REST client and container with the Beam job server jar and
+ * SDK client (for example when using the FlinkK8sOperator).
+ *
+ * <p>Eliminates the need to build jar files with materialized pipeline protos offline. Allows the
+ * driver program to access actual execution environment and services, on par with code executed by
+ * SDK workers.
+ *
+ * <p>The entry point starts the job server and provides the endpoint to the the driver program.
+ *
+ * <p>The external driver program constructs the Beam pipeline and submits it to the job service.
+ *
+ * <p>The job service defers execution of the pipeline to the plan environment and returns the
+ * "detached" status to the driver program.
+ *
+ * <p>Upon arrival of the job invocation, the entry point executes the runner, which prepares
+ * ("executes") the Flink job through the plan environment.
+ *
+ * <p>Finally Flink launches the job.
+ */
+public class FlinkPortableClientRunner {
+  private static final Logger LOG = LoggerFactory.getLogger(FlinkPortableClientRunner.class);
+  private static final String DRIVER_CMD_FLAGS = "--job_endpoint=%s";
```

Comment:
```suggestion
  private static final String JOB_ENDPOINT_FLAG = "--job_endpoint";
```

Reply:
This allows us to check for the args in the user command string. The `%s` would have to be appended in the format string.